### PR TITLE
Add first working version of a root_glob object

### PR DESCRIPTION
**Pull request by benkrikler**
This adds an implementation of `root_glob` which extends python's `glob` module to work over remote file systems.  It's based heavily on the way TChain handles wildcards, and as such has the same shortcoming, that only wild-cards in the basename are handled, and not in any directory name.

We can expand this in the future, to make things more like full globbing (which needs some recursion as python `glob` does) but my goal is to give us something that works for now.   I'll also put up a PR for this on rootpy so hopefully we can actually remove this in the future.

Should fix issue #16

### CODE REVIEW

**Comment by benkrikler** [comment]
Hmm, all checks have failed but the job log looks empty as far as I can tell. @kreczko, any ideas?  I checked that it was PEP-8 compliant before pushing it this time, so I don't think it's that...

File: cmsl1t/utils/root_glob.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        gSystem.FreeDirectory(directory)
+    return files
+
+
+def iglob(pathname):
+    for name in glob(pathname):
+        yield name
+
+
+if __name__ == "__main__":
```
Comment (kreczko) [inline cmsl1t/utils/root_glob.py]: Could you please make this into a unit test?

Comment (benkrikler) [inline cmsl1t/utils/root_glob.py]: I can do.  How do I unit-test files on an xrootd server, though?  Does the build system have access to this?  I'm not sure how I can fake such a thing either.

Comment (kreczko) [inline cmsl1t/utils/root_glob.py]: You could use the `mock` library to simulate a response, but you would need to patch the two gsystem calls. I will have a look and you can create an empty draft for now.

Comment (benkrikler) [inline cmsl1t/utils/root_glob.py]: Ok, I've added the draft, with tests to check that `root_glob.glob` performs the same as `glob` on local files, assuming the local files are in `data/`.

**Comment by kreczko** [comment]
thanks!

**Pull request merged**